kernel/src/hil/sensors.rs

```diff
@@ -289,3 +289,35 @@ pub trait DistanceClient {
     ///                If there was an error, this will be `Err(ErrorCode)`.
     fn callback(&self, distance: Result<u32, ErrorCode>);
 }
+
+/// A basic interface for a rain fall sensor
+pub trait RainFallDriver<'a> {
+    fn set_client(&self, client: &'a dyn RainFallClient);
+
+    /// Read the rain fall value from a sensor. The value is returned
+    /// via the `RainFallClient` callback.
+    ///
+    /// - `hours`: the number of hours of rainfall to report. 0
+    /// is the instant rainfall (if supported). Otherwise 1 to 24
```

Comment:
what does instant rainfall mean? Would Rainfall client return a rate rather than a length if 0 is passed?

Reply:
Yeah, good point. I guess it means the current rate, I removed that comment as nothing actually uses it